plugins/module_utils/network/sonic/config/users/users.py

```diff
@@ -152,7 +153,9 @@ def set_state(self, want, have):
         if not want:
             want = []
 
-        new_want = [{'name': conf['name'], 'role': conf['role']} for conf in want]
+        new_want = [{'name': conf['name'], 'role': conf['role'], 'ssh_key': conf['ssh_key']}
```

Comment:
Please add a comment here and also at the beginning of the handling for the "password" options (before the current line 167) to highlight the purpose of these two sections of configuration parsing.

```suggestion
                # Handle "role" and "ssh_key" options.
                new_want = [{'name': conf['name'], 'role': conf['role'], 'ssh_key': conf['ssh_key']}
```
and:

                                             * 
                                             * 
                                             * 
                            # Handle "password" and "update_password" options.

Reply:
Done

Reply:
Done

Reply:
The current (updated) version of this file doesn't contain the requested new comment preceding the "password" and "update_password" handling to clearly separate it from the preceding section (for role and ssh_key handling).

Please add a comment something like the following before the current line 168 in the "config" file:

`# Handle "password" and "update_password" options.`